enginetest/file_setup.go

```diff
@@ -0,0 +1,219 @@
+package enginetest
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	ast "github.com/dolthub/vitess/go/vt/sqlparser"
+)
+
+type ScriptHarness interface {
+	Setup() []setupSource
+}
+
+type setupSource interface {
+	Next() (bool, error)
+	Close() error
+	Data() Testdata
+}
+
+type Testdata struct {
```

Comment:
Not sure how much of this is a draft

But this struct should be used solely for parsing the test data files / code gen. We should have generated golang data structures checked into source corresponding to each of the textual setup files